File: win-prog/ProgWinCS/Taming the Mouse/MouseWeb/MouseWeb.cs
//---------------------------------------
// MouseWeb.cs © 2001 by Charles Petzold
//---------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class MouseWeb: Form
{
     Point ptMouse = Point.Empty;

     public static void Main()
     {
          Application.Run(new MouseWeb());
     }
     public MouseWeb()
     {
          Text = "Mouse Web";
          BackColor = SystemColors.Window;
          ForeColor = SystemColors.WindowText;
          ResizeRedraw = true;
     }
     protected override void OnMouseMove(MouseEventArgs mea)
     {
          Graphics grfx = CreateGraphics();

          DrawWeb(grfx, BackColor, ptMouse);
          ptMouse = new Point(mea.X, mea.Y);
          DrawWeb(grfx, ForeColor, ptMouse);

          grfx.Dispose();
     }
     protected override void OnPaint(PaintEventArgs pea)
     {
          DrawWeb(pea.Graphics, ForeColor, ptMouse);
     }
     void DrawWeb(Graphics grfx, Color clr, Point pt)
     {
          int cx  = ClientSize.Width;
          int cy  = ClientSize.Height;
          Pen pen = new Pen(clr);

          grfx.DrawLine(pen, pt, new Point(         0,          0));
          grfx.DrawLine(pen, pt, new Point(    cx / 4,          0));
          grfx.DrawLine(pen, pt, new Point(    cx / 2,          0));
          grfx.DrawLine(pen, pt, new Point(3 * cx / 4,          0));
          grfx.DrawLine(pen, pt, new Point(    cx    ,          0));
          grfx.DrawLine(pen, pt, new Point(    cx    ,     cy / 4));
          grfx.DrawLine(pen, pt, new Point(    cx    ,     cy / 2));
          grfx.DrawLine(pen, pt, new Point(    cx    , 3 * cy / 4));
          grfx.DrawLine(pen, pt, new Point(    cx    ,     cy    ));
          grfx.DrawLine(pen, pt, new Point(3 * cx / 4,     cy    ));
          grfx.DrawLine(pen, pt, new Point(    cx / 2,     cy    ));
          grfx.DrawLine(pen, pt, new Point(    cx / 4,     cy    ));
          grfx.DrawLine(pen, pt, new Point(         0,     cy    ));
          grfx.DrawLine(pen, pt, new Point(         0,     cy / 4));
          grfx.DrawLine(pen, pt, new Point(         0,     cy / 2));
          grfx.DrawLine(pen, pt, new Point(         0, 3 * cy / 4));
     }
}

File: win-prog/ProgWinVB/Taming the Mouse/MouseWeb/MouseWeb.vb
'-----------------------------------------
' MouseWeb.vb (c) 2002 by Charles Petzold
'-----------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class MouseWeb
    Inherits Form

    Private ptMouse As Point = Point.Empty

    Shared Sub Main()
        Application.Run(New MouseWeb())
    End Sub

    Sub New()
        Text = "Mouse Web"
        BackColor = SystemColors.Window
        ForeColor = SystemColors.WindowText
        ResizeRedraw = True
    End Sub

    Protected Overrides Sub OnMouseMove(ByVal mea As MouseEventArgs)
        Dim grfx As Graphics = CreateGraphics()
        DrawWeb(grfx, BackColor, ptMouse)
        ptMouse = New Point(mea.X, mea.Y)
        DrawWeb(grfx, ForeColor, ptMouse)
        grfx.Dispose()
    End Sub

    Protected Overrides Sub OnPaint(ByVal pea As PaintEventArgs)
        DrawWeb(pea.Graphics, ForeColor, ptMouse)
    End Sub

    Private Sub DrawWeb(ByVal grfx As Graphics, ByVal clr As Color, _
                                                ByVal pt As Point)
        Dim cx As Integer = ClientSize.Width
        Dim cy As Integer = ClientSize.Height
        Dim pn As New Pen(clr)

        grfx.DrawLine(pn, pt, New Point(0, 0))
        grfx.DrawLine(pn, pt, New Point(cx \ 4, 0))
        grfx.DrawLine(pn, pt, New Point(cx \ 2, 0))
        grfx.DrawLine(pn, pt, New Point(3 * cx \ 4, 0))
        grfx.DrawLine(pn, pt, New Point(cx, 0))
        grfx.DrawLine(pn, pt, New Point(cx, cy \ 4))
        grfx.DrawLine(pn, pt, New Point(cx, cy \ 2))
        grfx.DrawLine(pn, pt, New Point(cx, 3 * cy \ 4))
        grfx.DrawLine(pn, pt, New Point(cx, cy))
        grfx.DrawLine(pn, pt, New Point(3 * cx \ 4, cy))
        grfx.DrawLine(pn, pt, New Point(cx \ 2, cy))
        grfx.DrawLine(pn, pt, New Point(cx \ 4, cy))
        grfx.DrawLine(pn, pt, New Point(0, cy))
        grfx.DrawLine(pn, pt, New Point(0, cy \ 4))
        grfx.DrawLine(pn, pt, New Point(0, cy \ 2))
        grfx.DrawLine(pn, pt, New Point(0, 3 * cy \ 4))
    End Sub
End Class
